operator-common/src/main/java/io/strimzi/operator/common/model/PemAuthIdentity.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.common.model;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.strimzi.operator.common.Util;
+
+import java.io.ByteArrayInputStream;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Optional;
+
+/**
+ * Represents the identity used during TLS client authentication.
+ * This consists of an X509 end-entity certificate, corresponding private key, and a (possibly empty) chain of X509 intermediate CA certificates, all in PEM format.
+ */
+public class PemAuthIdentity {
+
+    private final byte[] privateKeyAsPemBytes;
+    private final byte[] certificateChainAsPemBytes;
+    private final String secretCertName;
+    private String secretName;
+    private String secretNamespace;
```

Comment:
Why do we need to keep this? We don't care about it when we have the identity, or?

Reply:
We use it for the exception if we fail to parse the data into a X509Certificate object

Reply:
I guess I wonder if we should cut our losses and simplify this and simply in some cases rely on the stacktrace?

Reply:
I at least found it useful when debugging why the tests failed, so I could see which secret had the corrupted certificate, so I think it is useful